force-app/main/default/classes/SOQL.cls

```diff
@@ -108,7 +139,13 @@ public with sharing class SOQL {
      * @return A list of Opportunity records.
      */
     public static List<Opportunity> getOpportunitiesByNameKeyword(String keyword) {
-        return null; // Replace null with the variable you used to store the result
+        String searchPattern = '%' + keyword + '%';
```

Comment:
Well done using wildcards and the LIKE comparison operator to get partial matches in your query